### PR TITLE
ICE adapter debug window

### LINKED ISSUE #2579: Allow launching ice adapter debug window

### PR DESCRIPTION
**Pull request by 1-alex98**

### CODE REVIEW

**Comment by 1-alex98** [comment]
Fixes #2579

File: src/main/java/com/faforever/client/fa/relay/ice/IceAdapterImpl.java [hunk 1]
```diff
@@ -154,10 +155,16 @@ public void onGpgGameMessage(GpgOutboundMessageEvent event) {
 
       Path workDirectory = Path.of(nativeDir).toAbsolutePath();
 
+      var classpath = getBinaryName(workDirectory);
```
Comment (Sheikah45) [inline src/main/java/com/faforever/client/fa/relay/ice/IceAdapterImpl.java]: We don't really use `var` anywhere else so please use the full type 😝

**Review by Sheikah45 — COMMENTED** [review]
Saw something quick will do a full review later

**Comment by Geosearchef** [comment]
Euhm. Wow. Wtf? This works?
Thanks for single-handidly dispersing all my worriers about the ice adapter debug capabilities without modifying the actual adapter.

I tried upgrading the java version of it to integrate JFX, but ran into quite some issues.

Does this workaround work, given that the ice adapter compiled with java 8 and the jfx you use is from the client?

**Comment by Sheikah45** [comment]
I suspect that since the client java is running the ice adapter it is running at the jvm15 level so as long as the class names are the same it should run.

**Comment by 1-alex98** [comment]
I tried it works flawlessly

File: src/main/java/com/faforever/client/fa/relay/ice/IceAdapterImpl.java (hunk 1 above)
Comment (1-alex98) [reply src/main/java/com/faforever/client/fa/relay/ice/IceAdapterImpl.java]: Ok... I did use it a lot lately. And I like it. Especially now Intellij always shows the type of vars

**Comment by 1-alex98** [comment]
> Euhm. Wow. Wtf? This works? Thanks for single-handidly dispersing all my worriers about the ice adapter debug capabilities without modifying the actual adapter.
> 
> I tried upgrading the java version of it to integrate JFX, but ran into quite some issues.
> 
> Does this workaround work, given that the ice adapter compiled with java 8 and the jfx you use is from the client?

jvm can execute older version byte code without any issues and also mix it with new byte code

File: src/main/java/com/faforever/client/fa/relay/ice/IceAdapterImpl.java [hunk 2]
```diff
@@ -237,6 +244,12 @@ public void onGpgGameMessage(GpgOutboundMessageEvent event) {
     return iceAdapterClientFuture;
   }
 
+  private String javaFXClassPathJars() {
+    final String classPathOfClient = System.getProperty("java.class.path");
+    final var split = List.of(classPathOfClient.split(";"));
```
Comment (Sheikah45) [inline src/main/java/com/faforever/client/fa/relay/ice/IceAdapterImpl.java]:
```suggestion
    final List<String> split = List.of(classPathOfClient.split(";"));
```

**Review by Sheikah45 — CHANGES_REQUESTED** [review]

File: src/main/java/com/faforever/client/fa/relay/ice/IceAdapterImpl.java (hunk 2 above)
Comment (1-alex98) [reply src/main/java/com/faforever/client/fa/relay/ice/IceAdapterImpl.java]: meh XD

File: src/main/java/com/faforever/client/fa/relay/ice/IceAdapterImpl.java (hunk 1 above)
Comment (Sheikah45) [reply src/main/java/com/faforever/client/fa/relay/ice/IceAdapterImpl.java]: Yeah understood although I don't know how I feel about it yet

**Review by Sheikah45 — APPROVED** [review]

**Pull request merged**